src/imageLoader/internal/xhrRequest.js

```diff
@@ -21,12 +21,18 @@ function xhrRequest(url, imageId, defaultHeaders = {}, params = {}) {
     const xhr = new XMLHttpRequest();
 
     xhr.open('get', url, true);
-    const beforeSendHeaders = options.beforeSend(
-      xhr,
-      imageId,
-      defaultHeaders,
-      params
-    );
+    
+    let beforeSendHeaders = {};
+    try{
+      beforeSendHeaders = options.beforeSend(
```

Comment:
This is still a synchronous call.  This is also not the current version - there is an options.open method that runs before on line 23.  Since this is already a promise, it should be easy to change this into an async function on line 4, then return the result directly down below.

Reply:
Yes it seems to be outdated base, @erasromani can you please rebase?